source/grid/manifold.cc

```diff
@@ -274,7 +336,8 @@ Point<spacedim>
 Manifold<dim, spacedim>::
 get_new_point_on_line (const typename Triangulation<dim, spacedim>::line_iterator &line) const
 {
-  return get_new_point (get_default_quadrature(line));
+  const std::pair<std::vector<Point<spacedim> >, std::vector<double> > points_weights(get_default_points_and_weights(line));
+  return get_new_point (points_weights.first,points_weights.second);
```

Comment:
Here and in the two two functions below, we still allocate memory. At some point, we might get around by re-using some data arrays in thread local storage, but I think for now you already reduce the number of memory allocations from 5 to 2 or so.